syntax = "proto3";

package auth;

// Mensagem para registrar um usuário
message RegisterRequest {
    string email = 1;
    string password = 2;
}

// Mensagem para fazer login
message LoginRequest {
    string email = 1;
    string password = 2;
}

// Mensagem de resposta
message AuthResponse {
    bool success = 1;
    string message = 2;
    string token = 3;
}

// Mensagem para listar usuários (não precisa de campos, então está vazio)
message ListUsersRequest {
    string token = 1;
}

// Mensagem de resposta para listar usuários (contém apenas o e-mail)
message User {
    string email = 1;
}



// Serviço de autenticação
service AuthService {
    rpc Register(RegisterRequest) returns (AuthResponse);
    rpc Login(LoginRequest) returns (AuthResponse);
    rpc ListUsers(ListUsersRequest) returns (stream User); // Novo método para listar usuários
}
